### PR TITLE
[DLS-8363] Adding case classes for Variations Contact and Variations …

### PR DESCRIPTION
**Pull request by JakeReid2020**
…Personal Details

### CODE REVIEW

File: app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                                      emailAddress: Option[String] = None) {
+  def nonEmpty: Boolean = Seq(name, position, telephoneNumber, emailAddress).flatten.nonEmpty
+}
+
+object VariationsPersonalDetails {
+
+  def apply(userAnswers : UserAnswers, subscription : RetrievedSubscription):Option[VariationsPersonalDetails] = {
+    userAnswers.get(UpdateContactDetailsPage).flatMap{ answers =>
+      val name = if(answers.fullName == subscription.contact.name.getOrElse(None)){None} else {Some(answers.fullName)}
+      val position = if(answers.position == subscription.contact.positionInCompany.getOrElse(None)){None} else {Some(answers.position)}
```
Comment (ChrisHillCap) [inline app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala:36]: should these be 
subscription.contact.name.getOrElse("")
subscription.contact.positionInCompany.getOrElse("")
i don't think answers.fullname and answers.position are optional so its comparing unrelated types?

File: app/models/updateRegisteredDetails/BusinessAddress.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.updateRegisteredDetails
+
+import play.api.libs.json._
+
```
Comment (ChrisHillCap) [inline app/models/updateRegisteredDetails/BusinessAddress.scala:20]: is this needed

**Review by ChrisHillCap — APPROVED** [review]

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]
see previous comments

File: app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala (hunk 1 above)
Comment (JakeReid2020) [reply app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala:36]: in retrieved subscription these are optional

Comment (JakeReid2020) [reply app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala:36]: and optional in VariationsPersonalDetails

Comment (ChrisHillCap) [reply app/models/updateRegisteredDetails/Submission/VariationsPersonalDetails.scala:36]: not optional in user answers though
userAnswers.fullName (this is mandatory) comparing it to the subscription get or else none
so potentially youre comparing user answers.fullname to None,
it should be userAnswers.fullname compared to subscription answer OR ""

**Review by ChrisHillCap — APPROVED** [review]

**Pull request merged**